hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -1861,7 +1861,7 @@ private class Emit(
         val moveArgs = refs.map { ref =>
           Code(ref.m := ref.tempM, ref.v := ref.tempV.load())
         }
-        EmitTriplet(Code(Code(storeTempArgs ++ moveArgs), jump.tcode[Unit]), const(false), PValue._empty)
+        EmitTriplet(Code(Code(storeTempArgs ++ moveArgs), jump.tcode[Unit]), const(false), PValue(pt, Code._empty))
```

Comment:
So Recur must be in tail position.  `pt` is the correct type, but the value should never be used.  I'd say `null` is more appropriate than `Code._empty`.

Reply:
What is the semantic difference? I read the EmitTriplet comment that stated that for void triplets value is Code._empty, and 

```scala
case Void() =>
        EmitTriplet(Code._empty, const(false), PValue(pt, Code._empty))
```

I don't full understand when we want something that follows acts like Void in code-generated code vs something that has a value that is always null.

Reply:
Recur doesn't have type void.  It has the same type as the loop.  Void type here was just wrong.

The value is just a placeholder because it isn't used.  `null` is a good choice because it crashes if it is used.

Reply:
> same type as the loop

as the body of the loop.